### ISSUE TITLE
Best option to train on custom dataset

### ISSUE THREAD

**Issue by hussienWehbi** [AlexeyAB/darknet#7372]
Hi @AlexeyAB 
I still have a conflict in the difference between transfer learning concept and fine-tuning concept , can you help me please ? also if I have a different dataset from those found in Image-Net (not trained in the pre-trained models) and I want to train all layers from scratch in yolov3-tiny ,is this better ? and how can I do that ? 
thank you very much

**Comment by stephanecharette** [comment]
Is this what you are looking for?  https://www.ccoderun.ca/programming/darknet_faq/#training_command

**Comment by hussienWehbi** [comment]
Hi @stephanecharette 
Actually I am using yolov3-tiny-prn.cfg architecture as @AlexeyAB said in one of the issues and its working fine , but I read that if the custom object I want to train is not found in COCO or Image-Net (datasets that the pre-trained models use) its better to train the early layers not just the final layers , so I am not sure if this is true and do not know how to do this if you can help me 
thanks

**Comment by stephanecharette** [comment]
If you are training your own custom network, I would start here:  https://www.ccoderun.ca/programming/darknet_faq/#configuration_template

If you are just getting started, you may find it beneficial to read through the entire FAQ.

**Comment by hussienWehbi** [comment]
@stephanecharette Thank you very much
i have already tried yolov3-tiny_3l.cfg with my own anchors and it worked fine but with much time for processing than tiny-prn and since i have an edge device with restricted ability (CPU core i3 and later on RapberryPI) time(ms) means a lot for me, and because accuracy is also very important i am trying to train from scratch so that i will get the best accuracy possible.
if you have an experience i hope you help me
Best regards

**Comment by stephanecharette** [comment]
Yes, I have experience.  That is why I posted those links showing 1) how to train from scratch, and 2) what configuration file to use.  My experience is also with Raspberry Pi, Jetson Nano, etc.

**Comment by hussienWehbi** [comment]
if i want a model to give me acceptable fps on edge devices but keeping a good accuracy to detect objects (can be small or big objects) , which one of tiny-prn & tiny_3l should i use ?
and if there exist a different network architecture (resnet-50, efficient-net, squeeze-net) that gives me better result than the two mentioned above i will be very thankful if you give me an idea
@stephanecharette again thank you for helping me

**Comment by stephanecharette** [comment]
I've already answered these questions.  See the links I posted above.

**Comment by ridvanozdemir** [comment]
any recommendation for using new architecture, or modifying cfg file for gray scale images?
